libraries/Microsoft.Bot.Builder.Dialogs/Prompts/ActivityPrompt.cs

```diff
@@ -117,8 +118,8 @@ public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext d
 
             // Perform base recognition
             var instance = dc.ActiveDialog;
-            var state = (IDictionary<string, object>)instance.State[PersistedState];
-            var options = (PromptOptions)instance.State[PersistedOptions];
+            var state = instance.State.GetTypedValue<IDictionary<string, object>>(PersistedState);
+            var options = instance.State.GetTypedValue<PromptOptions>(PersistedOptions);
```

Comment:
Talking with @johnataylor and we realized this would break current bots which are using a custom PromptOptions.

Reply:
Great catch.  I modified PromptOptions to have JsonExtensionDAta so properties will carry forward.  If there are custom PromptOptions and theyy then call TryGetTypedValue() it will be correctly coerced to the custom class type.  

---
In reply to: [464595322](https://github.com/microsoft/botbuilder-dotnet/pull/4369#discussion_r464595322) [](ancestors = 464595322)